kong/tools/timestamp.lua

```diff
@@ -5,56 +5,71 @@
 -- @module kong.tools.timestamp
 
 local luatz = require "luatz"
-local _M = {}
+local tz_time = luatz.time
+local tt_from_timestamp = luatz.timetable.new_from_timestamp
+local tt = luatz.timetable.new
+local math_floor = math.floor
 
 --- Current UTC time
--- @return UTC time
-function _M.get_utc()
-  return math.floor(luatz.time()) * 1000
+-- @return UTC time in milliseconds since epoch, but with SECOND precision.
+local function get_utc()
+  return math_floor(tz_time()) * 1000
 end
 
-function _M.get_timetable(now)
-  local timestamp = now and now or _M.get_utc()
-  if #tostring(timestamp) == 13 then
-    timestamp = timestamp / 1000
-  end
-  return luatz.timetable.new_from_timestamp(timestamp)
+--- Current UTC time
+-- @return UTC time in milliseconds since epoch.
+local function get_utc_ms()
+  return tz_time() * 1000
 end
 
---- Creates a timestamp
--- @param now (optional) Time to generate a timestamp from, if omitted current UTC time will be used
--- @return Timestamp table containing fields; second, minute, hour, day, month, year
-function _M.get_timestamps(now)
-  local timetable = _M.get_timetable(now)
+-- setup a validation value, any value above this is assumed to be in MS 
+-- instead of S (a year value beyond the year 20000), it assumes current times
+-- as in 2016 and later.
+local ms_check = tt(20000 , 1 , 1 , 0 , 0 , 0):timestamp()
```

Comment:
20000, dang, that's a golden era, can't wait to be there (https://en.wikipedia.org/wiki/Far_future_in_science_fiction_and_popular_culture)